Task 1. (Cargo) operator== & Task 2. (Cargo) Constructor

kasiasieradzka opened the pull request:
resolve #2 resolve #3

kasiasieradzka committed:
zadanie1 i zadanie2

Kasperice commented:
Looks good for me. It will be easier for me to merge that after #11 PR which implements Ship and Cargo from classes.

Kasperice commented on shm/cargo.cpp:
```diff
@@ -0,0 +1,16 @@
+
+std::string getName() const { return name_; }
+size_t getAmount() const { return amount_; }
+size_t getBase_price() const { return base_price_; }
```
Let's do not mix two naming conventions
```suggestion
size_t getBasePrice() const { return base_price_; }
```

Kasperice commented on shm/cargo.hpp:
```diff
@@ -0,0 +1,21 @@
+  private: 
+     std::string name_;
+     size_t amount_;
+     size_t base_price_;
```
We decided on Discord to use camelCase for functions and variables

```suggestion
     size_t basePrice_;
```

Kasperice replied:
Please change it everywhere

Kasperice commented on shm/cargo.hpp:
```diff
@@ -0,0 +1,21 @@
+     size_t base_price_;
+
+  public: 
+     Cargo();
```
As I see sometimes your indentation is 4 sometimes 5, change it to 4.

kasiasieradzka committed:
Update cargo.hpp

g00fer commented on shm/cargo.hpp:
```diff
@@ -0,0 +1,20 @@
+class Cargo {
+  private: 
+     std::string name_;
+     size_t amount_;
```
It is a wrong use of size_t. size_t is  for indexing arrays, vectors etc. https://en.cppreference.com/w/cpp/types/size_t
Ideally type of amount should be the measurement units.
so ideally: Kilograms amount_;   Then you can do: cargo1.amount_ + cargo2.amount_  But you cannot do cargo2.amount_ + cargo1.base_price_

g00fer commented on shm/cargo.hpp:
```diff
@@ -0,0 +1,20 @@
+  private: 
+     std::string name_;
+     size_t amount_;
+     size_t basePrice_;
```
Use normal type here (int, unisigned int , uint32_t or typedef (e.g Gold))

Kasperice replied:
For amount and basePrice (and similar ones) i used uint8_t it should be enough for now

Kasperice replied:
@kasiasieradzka sorry for confusion, my usage of uint8_ was not the best idea. As @g00fer said uint32_t will be better. If you want you can change that, if not I will change it during merge

kasiasieradzka replied:
@Kasperice, it fixed

Kasperice replied:
thanks :)

kasiasieradzka committed:
Update2 cargo.hpp

kasiasieradzka committed:
Update2 cargo.cpp

Pawellos commented on shm/cargo.cpp:
```diff
@@ -0,0 +1,15 @@
+#include "cargo.hpp"
+
+Cargo();
```
I think it will not complie, there should be scope, like this:
Cargo::Cargo();

kasiasieradzka replied:
But I think we use Cargo::Cargo(); when we are outside of class, inside we don't need it

Kasperice replied:
Everywhere I saw for constructor is ClassName::ClassName(). As I remember even VSC underline it if I did not use ClassName::ClassName(), but someone more experienced could clarify

kasiasieradzka replied:
Yes, I mixed up everything... Give me few minutes to fix it

Pawellos commented on shm/cargo.cpp:
```diff
@@ -0,0 +1,15 @@
+{}
+~Cargo() {;} 
+
+std::string getName() const { return name_; }
+uint8_t getAmount() const { return amount_; }
+uint8_t getBase_price() const { return base_price_; }
```
every where with methods should be :

std::string Cargo::getName() const { return name_; }
uint8_t Cargo::getAmount() const { return amount_; }
uint8_t Cargo::getBase_price() const { return base_price_; }

in my opinion

g00fer reviewed (changes requested):
Please make sure it is compililing.

kasiasieradzka committed:
Update cargo.cpp

I applied the changes that you proposed, but I'm not sure if this should look like this

kasiasieradzka committed:
Once more update cargo.cpp

kasiasieradzka committed:
Update cargo.hpp

kasiasieradzka committed:
update(3) cargo.cpp

kasiasieradzka commented:
@g00fer can you tell me how compiling it on github?

kasiasieradzka committed:
Fixed cargo.cpp
